pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/PrimaryKey.java

```diff
@@ -37,7 +41,57 @@ public Object[] getValues() {
   }
 
   public byte[] asBytes() {
-    return SerializationUtils.serialize(_values);
+    int sizeInBytes = 0;
+    byte[][] cache = new byte[_values.length][];
+    for (int i = 0; i < _values.length; i++) {
+      Object value = _values[i];
+      if (value == null) {
```

Comment:
It should not be `null` or we cannot guarantee different primary key generate different bytes.